### DISCUSSION TITLE
How to specify an upstream dependency for dbt-based assets

### DISCUSSION THREAD

**Discussion by dpeng817** [dagster-io/dagster#13995] [Q&A] [answered]
When using auto-load technologies for loading dbt projects as assets, how can you specify upstream dependencies to those assets?

**Comment by slopp** [comment, marked as answer]
Dagster determines upstream dependencies for dbt-based assets by looking inside the dbt sources.yaml. Each table in sources.yaml is converted into to a Dagster asset key.

If a dbt model references a table named "raw_orders" inside a dbt source named "prod_db", then the Dagster asset corresponding to that dbt model will have an asset dependency with `AssetKey(["prod_db", "raw_orders"])`.

If you have an asset with that key in your Dagster deployment, then the UI will show the connection between those assets. If you Materialize them both, Dagster will wait for the raw_orders asset to complete before running the downstream dbt model.

You can use the `source_key_prefix` or `node_info_to_asset_key` functions to customize the asset keys that are created from dbt sources.

**Reply by sryza** [reply]
@slopp heads up that I updated the answer to reflect the fact that, perhaps a little counterintuitively, we don't automatically create `SourceAsset`s for dbt sources. And also that dbt sources correspond to sets of source tables, not single tables. Let me know if you think the revised answer is unclear and feel free to amend further.
